StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1500BracesForMultiLineStatementsMustNotShareLine.cs

```diff
@@ -288,11 +288,10 @@ private static void CheckBraceToken(SyntaxNodeAnalysisContext context, SyntaxTok
                     // Because the default Visual Studio code completion snippet for a do-while loop
                     // places the while expression on the same line as the closing brace, some users
                     // may want to allow that and not have SA1500 report it as a style error.
-                    if (context.Options.GetStyleCopSettings(context.CancellationToken).LayoutRules.AllowDoWhileOnClosingBrace)
+                    if (context.GetStyleCopSettings(context.CancellationToken).LayoutRules.AllowDoWhileOnClosingBrace)
                     {
-                        var openBracePreviousToken = openBraceToken.GetPreviousToken(includeZeroWidth: true);
-
-                        if (openBracePreviousToken.IsKind(SyntaxKind.DoKeyword))
+                        if (openBraceToken.Parent.IsKind(SyntaxKind.Block)
+                            && openBraceToken.Parent.Parent.IsKind(SyntaxKind.DoStatement))
```

Comment:
Since I don't know the syntax tree well enough, I just want to make sure, as I had similar code and this same question elsewhere, "Is using `.Parent.Parent` safe here?" We know that `openBraceToken.Parent` is not `null` and is of kind `Block`, but does that guarantee that `.Parent.Parent` will never be `null`? For example what if I have code like the following that uses some extra curly braces for scoping:
```
{ // Some braces for scoping
   var x = 0;
   do
   {
      x = 1;
   } while (x == 0);
}
```

Reply:
`IsKind` allows null, and `openBraceToken.Parent.IsKind(SyntaxKind.Block)` is only true if `openBraceToken.Parent` is not null.